openapi.yaml

```diff
@@ -0,0 +1,149 @@
+    Trades:
+      type: object
+      properties:
+        buy : { type: array, items: {$ref: '#/components/schemas/Trade' }}
+        sell : { type: array, items: {$ref: '#/components/schemas/Trade'}}
+      example:
+        {
+          "buy": [{ "trade_id":1234567, "price":"50.1", "base_volume":"0.1", "target_volume":"1", "trade_timestamp":"1700050000", "type":"buy" }],
+          "sell": [{ "trade_id":1234567, "price":"50.1", "base_volume":"0.1", "target_volume":"1", "trade_timestamp":"1700050000", "type":"sell" }]
+        }
+    Price:
+      type: object
+      properties:
+        coin_a_close: { type: number, format: double, description: Coin A close }
+        coin_a_high: { type: number, format: double, description: Coin B high }
+        coin_a_low: { type: number, format: double, description: Coin A low }
+        coin_a_open: { type: number, format: double, description: Coin A open }
+        coin_a_volume: {type: number, format: double, description: Coin A volume }
+        coin_b_close: { type: number, format: double, description: Coin B close }
+        coin_b_high: { type: number, format: double, description: Coin A high }
+        coin_b_low: { type: number, format: double, description: Coin B low }
+        coin_b_open: { type: number, format: double, description: Coin B open}
+        coin_b_volume: { type: number, format: double, description: Coin B volume }
+        count: { type: integer, description: Candlestick size }
+        time_period_start: { type: string , description: Trade timestamp, format: "date", example: "1985-04-12T23:00:00.00Z" }
+        time_period_end: { type: string , description: Trade timestamp, format: "date", example: "1985-04-12T23:15:00.00Z" }
+    Prices:
+      type: array
+      items: {$ref: '#/components/schemas/Price'}
+security:
+  - ApiKeyAuth: [read, write]
+tags: [{name: markets}, {name: orders}, {name: trades}, {name: prices}, {name: settings}]
+externalDocs:
+ description: Bot API Documentation
+ url: https://www.bot-api-docs.geniusyield.co/
+servers:
+  - url: https://localhost:4000/
+  - url: https://bot.api.geniusyield.co/
+paths:
+  /settings:
+    get:
+      tags: [settings]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Settings' }}}}}
+  /tickers:
+    get:
+      tags: [markets]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Tickers' }}}}}
+  /orderbook:
+    parameters:
+     - {in: query, name: ticker_id, required: true, schema: { type: string }}
+     - {in: query, name: depth, required: false, schema: { type: integer }}
+    get:
+      tags: [orders]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Orders' }}}}}
+    post:
```

Comment:
I am a bit torn if the POST and DELETE operations should be in the orderbook.

The orderbook name itself is also a product of our effort to be compatible with the Coin Gecko API requirements.

I would prefer to go purely RESTful and have plural resource names:
 - orders (for order book)
 - trades (for historical trades)
 - markets (for the trading pairs / tickers)